libs/experimental/langchain_experimental/comprehend_moderation/toxicity.py

```diff
@@ -134,76 +132,34 @@ def validate(
-                    BaseModerationActions,
-                )
-
-                toxicity_found = False
-                action = config.get("action", BaseModerationActions.STOP)
-                if action not in [
-                    BaseModerationActions.STOP,
-                    BaseModerationActions.ALLOW,
-                ]:
-                    raise ValueError("Action can either be stop or allow")
-
-                threshold = config.get("threshold", 0.5) if config else 0.5
-                toxicity_labels = config.get("labels", []) if config else []
-
-                if action == BaseModerationActions.STOP:
-                    for item in response["ResultList"]:
-                        for label in item["Labels"]:
-                            if (
-                                label
-                                and (
-                                    not toxicity_labels
-                                    or label["Name"] in toxicity_labels
-                                )
-                                and label["Score"] >= threshold
-                            ):
-                                toxicity_found = True
-                                break
-
-                if action == BaseModerationActions.ALLOW:
-                    if not toxicity_labels:
-                        warnings.warn(
-                            "You have allowed toxic content without specifying "
-                            "any toxicity labels."
-                        )
-                    else:
-                        for item in response["ResultList"]:
-                            for label in item["Labels"]:
-                                if (
-                                    label["Name"] in toxicity_labels
-                                    and label["Score"] >= threshold
-                                ):
-                                    toxicity_found = True
-                                    break
-
-                if self.callback and self.callback.toxicity_callback:
-                    if toxicity_found:
-                        self.moderation_beacon["moderation_status"] = "LABELS_FOUND"
-                    asyncio.create_task(
-                        self.callback.on_after_toxicity(
-                            self.moderation_beacon, self.unique_id
-                        )
-                    )
-                if toxicity_found:
-                    raise ModerationToxicityError
+            toxicity_found = False
+            threshold = config.get("threshold")
+            toxicity_labels = config.get("labels")
+
+            if not toxicity_labels:
+                for item in response["ResultList"]:
+                    for label in item["Labels"]:
+                        if label["Score"] >= threshold:
+                            toxicity_found = True
+                            break
```

Comment:
Nit: Better to create a model to represent this, and move the nesting of this computation to those models. For example:

```python
class Label(BaseModel, extra=Extra.allow):
    Score: int

    def istoxic(self, threshold: int) -> bool:
        return self.Score >= threshold

class Result(BaseModel, extra=Extra.allow):
    Labels: Sequence[Label] = []
    
    def istoxic(self, threshold: int) -> bool:
        for label in Labels:
            return True if label.istoxic(threshold)

        return False

class Response(BaseModel, extra=Extra.allow):
    ResultList: Sequence[Result]
    
    def istoxic(self, threshold: int) -> bool:
        for result in ResultList:
            return True if result.istoxic(threshold)

        return False

```

Then the code becomes:

```python
_response = Response.parse_obj(response)
if not toxicity_labels:
    toxicity_found = _response.istoxic(threshold)
```

Reply:
Will trace it in our next release and make changes accordingly